lfs/diff_index_scanner.go

```diff
@@ -0,0 +1,191 @@
+package lfs
+
+import (
+	"bufio"
+	"strconv"
+	"strings"
+
+	"github.com/git-lfs/git-lfs/errors"
+)
+
+// Status represents the status of a file that appears in the output of `git
+// diff-index`.
+//
+// More information about each of its valid instances can be found:
+// https://git-scm.com/docs/git-diff-index
+type Status rune
```

Comment:
@technoweenie I think this needs a better name, but I'm not sure what to call it. Exporting this field is necessary, since it'll be used from the `commands` package, but I think adding `lfs.Status` to the public API is too generic.

I was originally thinking of renaming this to be called `lfs.DiffIndexStatus` or `lfs.DiffIndexEntryStatus`. That got me thinking, what about moving these diff index changes to the `git` package, or a new `git/diffindex` package?

There are a few helpers that we'd have to move, like `lfs.startCommand`, but I don't think that'd be a problem.

What are your thoughts?